compiler/ballerina-lang/src/main/java/io/ballerina/projects/BuildToolResolution.java

```diff
@@ -0,0 +1,315 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (https://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.projects;
+
+import com.google.gson.JsonSyntaxException;
+import io.ballerina.projects.buildtools.CodeGeneratorTool;
+import io.ballerina.projects.buildtools.ToolContext;
+import io.ballerina.projects.environment.PackageLockingMode;
+import io.ballerina.projects.environment.ToolResolutionRequest;
+import io.ballerina.projects.internal.PackageDiagnostic;
+import io.ballerina.projects.internal.model.BuildJson;
+import io.ballerina.projects.util.BuildToolUtils;
+import io.ballerina.tools.diagnostics.Diagnostic;
+import org.ballerinalang.central.client.CentralAPIClient;
+import org.ballerinalang.central.client.exceptions.CentralClientException;
+import org.ballerinalang.central.client.model.ToolResolutionCentralRequest;
+import org.ballerinalang.central.client.model.ToolResolutionCentralResponse;
+import org.ballerinalang.toml.exceptions.SettingsTomlException;
+import org.wso2.ballerinalang.util.RepoUtils;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.ServiceLoader;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static io.ballerina.projects.util.ProjectConstants.BUILD_FILE;
+import static io.ballerina.projects.util.ProjectConstants.DEFAULT_VERSION;
+import static io.ballerina.projects.util.ProjectUtils.getAccessTokenOfCLI;
+import static io.ballerina.projects.util.ProjectUtils.initializeProxy;
+import static io.ballerina.projects.util.ProjectUtils.readBuildJson;
+
+/**
+ * {@code BuildToolResolution} Model for resolving tool dependencies.
+ *
+ * @since 2201.9.0
+ */
+public class BuildToolResolution {
+    private static final String PACKAGE_NAME_PREFIX  = "tool_";
+    private final PackageContext packageContext;
+    private final List<BuildTool> resolvedTools = new ArrayList<>();
```

Comment:
Let's do the initialization in the constructor

Reply:
Fixed with 9c900ee382a0ef596c5cbe9e2298352bc5cbfa89